misk-feature-testing/src/main/kotlin/misk/feature/testing/FakeFeatureFlags.kt

```diff
@@ -77,76 +77,149 @@ class FakeFeatureFlags @Inject constructor(private val moshi : Provider<Moshi>)
-    override(feature, value, T::class.java)
+    overrideKeyJson(feature, KEY, value, defaultAttributes)
   }
 
-  fun overrideKey(feature: Feature, key: String, value: Boolean) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Boolean,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Boolean>(feature, key, value, attributes)
 
-  fun overrideKey(feature: Feature, key: String, value: Int) {
-    overrides[MapKey(feature, key)] = value
-  }
 
-  fun overrideKey(feature: Feature, key: String, value: String) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Int,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Int>(feature, key, value, attributes)
 
-  fun overrideKey(feature: Feature, key: String, value: Enum<*>) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: String,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<String>(feature, key, value, attributes)
 
-  fun <T> overrideKey(feature: Feature, key: String, value: T, clazz : Class<T>) {
-    overrides[MapKey(feature, key)] = { moshi.get().adapter(clazz).toSafeJson(value) }
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Enum<*>,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Enum<*>>(feature, key, value, attributes)
 
-  inline fun <reified T> overrideKeyJson(feature: Feature, key: String, value: T) {
-    overrideKey(feature, key, value, T::class.java)
+  fun <T> overrideKey(
+    feature: Feature,
+    key: String,
+    value: T,
+    attributes: Attributes = Attributes()
+  ) {
+    val mapKey = MapKey(feature, key)
+    overrides[mapKey] = overrides[mapKey] ?: PriorityQueue()
+    overrides[mapKey]!!.add(MapValue(
+        order = overrides[mapKey]!!.size + 1,
+        attributes = attributes,
+        value = value as Any))
+  }
+
+  inline fun <reified T> overrideKeyJson(
+    feature: Feature,
+    key: String,
+    value: T,
+    attributes: Attributes = defaultAttributes
+  ) {
+    val jsonValue = { moshi.get().adapter(T::class.java).toSafeJson(value) } as Any
+    overrideKey(feature, key, jsonValue, attributes)
   }
 
   fun reset() {
     overrides.clear()
   }
 
-  private data class MapKey(val feature: Feature, val key: String? = null)
+  private data class MapKey(
+    val feature: Feature,
+    val key: String = KEY
+  )
+
+  /**
+   * Data class that holds the override value provided for the given [feature, key, attributes].
+   */
+  private data class MapValue(
+    val order: Int = 1, //Order is used to pick the latest provided map value.
+    val attributes: Attributes = defaultAttributes,
+    val value: Any
+  ) : Comparable<MapValue> {
+    override fun compareTo(other: MapValue) = when (attributes.text.size) {
```

Comment:
I find this control flow really awkward to follow because attributes.text.size shows up in 3 places. Simpler to go member-by-member?
```
val attributeSizeCompare = -attributes.text.size.compareTo(other.attributes.text.size)
if (attributeSizeCompare != 0) return attributeSizeCompare

return -order.compareTo(other.order)
```

Reply:
will change.